php/php-codesniffer-standard/VIISON/Sniffs/Methods/UnitTestMethodNameSniff.php

```diff
@@ -0,0 +1,161 @@
+<?php
+namespace VIISON\StyleGuide\PHPCS\Standards\VIISON\Sniffs\Methods;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+/**
+ * Finds methods whos method name does not match their testdox description.
+ */
+class UnitTestMethodNameSniff implements Sniff
+{
+    const METHOD_NAME_DOES_NOT_EQUAL_TESTDOX = 'Unit test method "%s" does not equal testdox. Expected "%s"';
+
+    /**
+     * @inheritdoc
+     */
+    public function register()
+    {
+        return [
+            T_DOC_COMMENT_OPEN_TAG,
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function process(File $phpcsFile, $stackPointer)
+    {
+        $tokens = $phpcsFile->getTokens();
+        $testdox = $this->getTestdox($tokens, $stackPointer);
+        if ($testdox === null) {
+            return;
+        }
+
+        $actualMethodName = $this->getMethodName($tokens, $stackPointer);
+        if ($actualMethodName === null) {
+            return;
+        }
+
+        $expectedMethodName = $this->getExpectedMethodNameFromTestdox($testdox);
+        if ($actualMethodName === $expectedMethodName) {
+            return;
+        }
+
+        $fixMethodName = $phpcsFile->addFixableError(sprintf(
+            self::METHOD_NAME_DOES_NOT_EQUAL_TESTDOX,
+            $actualMethodName,
+            $expectedMethodName
+        ), $stackPointer, 'EqualsTestdox');
+
+        if ($fixMethodName === true) {
+            $phpcsFile->fixer->beginChangeset();
+            $phpcsFile->fixer->replaceToken($stackPointer, $expectedMethodName);
+            $phpcsFile->fixer->endChangeset();
+        }
+    }
+
+    private function getMethodName($tokens, &$index)
+    {
+        $expectedTokens = [
+            'T_PUBLIC',
+            'T_FUNCTION',
+        ];
+
+        while (count($tokens) > $index) {
+            $token = $tokens[$index];
+            $index += 1;
+
+            if ($token['type'] === 'T_WHITESPACE') {
+                continue;
+            }
+
+            if (count($expectedTokens) > 0 && $token['type'] === $expectedTokens[0]) {
+                array_shift($expectedTokens);
+                continue;
+            }
+
+            if ($token['type'] === 'T_STRING') {
+                // The index was already increased by 1 and is currently pointing to the next token. Decrease the
+                // index so that it points to the current token.
+                $index -= 1;
+
+                return $token['content'];
+            } else {
+                return null;
+            }
+        }
+    }
+
+    private function getExpectedMethodNameFromTestdox($testdox)
+    {
+        $words = preg_split('/[^a-zA-Z0-9]/', $testdox);
+        $methodName = 'test_';
+        foreach ($words as $index => $word) {
+            if (mb_strlen($word) === 0) {
+                continue;
+            }
+
+            $firstLetter = mb_substr($word, 0, 1);
+            $followingLetters = mb_substr($word, 1);
```

Comment:
I feel like there's a call to `mb_strtolower` missing here (and a corresponding test case).